--- a/core/upstream/dolphin_web_core.cpp
+++ b/core/upstream/dolphin_web_core.cpp
@@ -38,6 +38,7 @@
 #include "Core/HW/EXI/EXI_Device.h"
 #include "Core/HW/GCKeyboard.h"
 #include "Core/HW/GCPad.h"
+#include "Core/HW/SI/SI_Device.h"
 #include "Core/HW/SystemTimers.h"
 #include "Core/Host.h"
 #include "Core/PowerPC/PowerPC.h"
@@ -593,6 +594,8 @@
   Config::SetBase(Config::MAIN_CPU_CORE, s_cpu_core);
   Config::SetBase(Config::MAIN_CPU_THREAD, s_cpu_thread);
   Config::SetBase(Config::MAIN_SKIP_IPL, true);
+  for (int port = 0; port < 4; ++port)
+    Config::SetBase(Config::GetInfoForSIDevice(port), SerialInterface::SIDEVICE_GC_CONTROLLER);
   Config::SetBase(Config::MAIN_DSP_HLE, true);
   Config::SetBase(Config::MAIN_DSP_THREAD, false);
   Config::SetBase(Config::MAIN_FAST_DISC_SPEED, true);
